src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -5941,7 +5941,8 @@ void CodeGen::genCompareInt(GenTree* treeNode)
 
         // Optimize "x<0" and "x>=0" to "x>>31" if "x" is not a jump condition and in a reg.
         // Morph/Lowering are responsible to rotate "0<x" to "x>0" so we won't handle it here.
-        if ((targetSize >= 4) && (op1Size >= 4) && (targetReg != REG_NA) && tree->OperIs(GT_LT, GT_GE))
+        if ((targetSize >= 4) && (op1Size >= 4) && (targetReg != REG_NA) && tree->OperIs(GT_LT, GT_GE) &&
```

Comment:
Sorry, I have not reviewed the original PR, but was it supported to be
```
emitAttr targetSize = emitActualTypeSize(tree->TypeGet()); // tree, not op1
emitAttr op1Size    = emitActualTypeSize(op1Type); // can use op1Type, but it is a nit.
```
?

and why do we have such restrictions? The actual type can't be less than 4, can it?

Reply:
Also should we condition this under `optimizationEnabled`? Seems odd to optimize this if we're not optimizing in general.

Reply:
@sandreenko - good catch, and sorry I missed it. In particular, the first two checks will never fail, because `emitActualTypeSize` will always return at least 4 for an integral type. I believe that they are not needed.

I would agree with @AndyAyersMS that this should be guarded by `compiler->opts.OptimizationEnabled()`

Reply:
Fixed, sorry it was a leftover from my previous commits in that PR, does it make sense to keep an assert?